internal/install/install.go

```diff
@@ -17,30 +17,34 @@ import (
 	"gopkg.in/yaml.v2"
 
 	"github.com/newrelic/newrelic-cli/internal/credentials"
+	"github.com/newrelic/newrelic-client-go/newrelic"
 )
 
-func install(configFiles []string) error {
+func install(client *newrelic.NewRelic) error {
+	ctx := getSignalContext()
+	rf := newServiceRecipeFetcher(&client.NerdGraph)
+	pf := newRegexProcessFilterer(rf)
+
 	// Execute the discovery process.
 	log.Debug("Running discovery...")
-	var d discoverer = new(psUtilDiscoverer)
-	manifest, err := d.discover()
+	var d discoverer = newPSUtilDiscoverer(pf)
+	manifest, err := d.discover(ctx)
 	if err != nil {
 		return err
 	}
 
-	log.Debugf("manifest: %+v", manifest)
-
 	// Retrieve the relevant recipes.
 	log.Debug("Retrieving recipes...")
-	var f recipeFetcher = new(yamlRecipeFetcher)
-	recipes, err := f.fetch(configFiles, manifest)
+	// TODO: pass context into this method.  The client will need to be updated
```

Comment:
We will want to pass context into the fetch methods here, but the underlying client does not yet support it.